cmd/libs/go2idl/client-gen/README.md

```diff
@@ -0,0 +1,23 @@
+Client-gen is an automatic tool that generates [clientset](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/client-package-structure.md#high-level-client-sets) based on API types. This doc introduces the use the client-gen, and the release cycle of the generated clientsets.
+
+## Using client-gen
+
+The workflow include four steps:
+- Marking API types with tags: in `pkg/apis/${GROUP}/${VERSION}/types.go`, mark the types (e.g., Pods) that you want to generate clients for with the `// +genclient=true` tag.  If the resource associated with the type is not namespace scoped (e.g., PersistentVolume), you need to append the `nonNamespaced=true` tag as well.
```

Comment:
The current implementation has this assumption. I'm willing to improve it.

The info needed by client-gen is a list of files that contains the types, and their respective group/version. How about we provide two sets of input flags?
- For the k8s built-in types, because our directories are structured, client-gen can infer the path to types.go from group/version, so user only need to specify the group/version
- For third-party apis, how about we require user to specify a list of pairs of `<path to types.go, group/version>`? Then the types.go can live anywhere.